### PR TITLE
Reorganize unit tests

### PR DESCRIPTION
**Pull request by kuhar**
Now that we have a number of unit tests and have better idea what dependencies
are necessary, make unit test organization more consistent.

-  Make all unit tests link with `llpc_standalone_compiler` for simplicity.
-  Make test directory names match implementation directories.
-  Remove the placeholder test.

### CODE REVIEW

File: llpc/unittests/CMakeLists.txt [hunk 1]
```diff
@@ -54,7 +54,7 @@ function(llpc_add_unittest_impl test_suite test_name)
   set(outdir ${CMAKE_CURRENT_BINARY_DIR}/${CMAKE_CFG_INTDIR})
```
Comment (vettoreldaniele) [inline llpc/unittests/CMakeLists.txt:54]: (I can't comment on the line outside of this change)
Can we rename `llpc_add_unittest_impl` to `add_llpc_unittest_impl` to match the public name `add_llpc_unittest` used elsewhere?

Comment (kuhar) [reply llpc/unittests/CMakeLists.txt:54]: Done

**Comment by kuhar** [comment]
V1. Rebase and rename `llpc_add_unittest_impl` to `add_llpc_unittest_impl`.

**Review by vettoreldaniele — APPROVED** [review]
LGTM

**Comment by amdvlk-admin** [comment]
### Test summary for commit 42c017aa9e057340abe498663e97d0989808e612
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>PAL: dev 44abc48d199dd28369bf7415eee1751c4bf616d2</li>
<li>SPVGEN: dev 580ade71d630a304805e9468ef22ad263c37d137</li>
<li>XGL: dev b8ed024d536d1bebb35927a0f4f470dca2f2dfc6</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev cdd74831d061bf65fdf5af4a0b6ce99fe2c558a9</li>
</ul></details>

<details><summary>CTS tests (Failed: 0/163959)</summary><ul><li>Built with version 1.2.5.2</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 32354/54653 (59.2%)</li>
<li>Failed: 0/54653 (0.0%)</li>
<li>Not Supported: 22299/54653 (40.8%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Timeout or crash</li>

</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 32530/54653 (59.5%)</li>
<li>Failed: 0/54653 (0.0%)</li>
<li>Not Supported: 22123/54653 (40.5%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx103</summary><ul>

<li>Passed: 35511/54653 (65.0%)</li>
<li>Failed: 0/54653 (0.0%)</li>
<li>Not Supported: 19142/54653 (35.0%)</li>
</ul></details>


</ul></details>

**Comment by amdvlk-admin** [comment]
### Test summary for commit c6a1afe5756985a92f57dcbf701dc5c212553ffb
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>PAL: dev 44abc48d199dd28369bf7415eee1751c4bf616d2</li>
<li>SPVGEN: dev 580ade71d630a304805e9468ef22ad263c37d137</li>
<li>XGL: dev b8ed024d536d1bebb35927a0f4f470dca2f2dfc6</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev cdd74831d061bf65fdf5af4a0b6ce99fe2c558a9</li>
</ul></details>

<details><summary>CTS tests (Failed: 0/163959)</summary><ul><li>Built with version 1.2.5.2</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 32354/54653 (59.2%)</li>
<li>Failed: 0/54653 (0.0%)</li>
<li>Not Supported: 22299/54653 (40.8%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Timeout or crash</li>

</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 32530/54653 (59.5%)</li>
<li>Failed: 0/54653 (0.0%)</li>
<li>Not Supported: 22123/54653 (40.5%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx103</summary><ul>

<li>Passed: 35511/54653 (65.0%)</li>
<li>Failed: 0/54653 (0.0%)</li>
<li>Not Supported: 19142/54653 (35.0%)</li>
</ul></details>


</ul></details>

**Comment by amdvlk-admin** [comment]
### Test summary for commit b93327b1872b01d1eb0de4497371206657074fec
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>PAL: dev 44abc48d199dd28369bf7415eee1751c4bf616d2</li>
<li>SPVGEN: dev 580ade71d630a304805e9468ef22ad263c37d137</li>
<li>XGL: dev b8ed024d536d1bebb35927a0f4f470dca2f2dfc6</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev cdd74831d061bf65fdf5af4a0b6ce99fe2c558a9</li>
</ul></details>

<details><summary>CTS tests (Failed: 0/163959)</summary><ul><li>Built with version 1.2.5.2</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 32354/54653 (59.2%)</li>
<li>Failed: 0/54653 (0.0%)</li>
<li>Not Supported: 22299/54653 (40.8%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Timeout or crash</li>

</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 32530/54653 (59.5%)</li>
<li>Failed: 0/54653 (0.0%)</li>
<li>Not Supported: 22123/54653 (40.5%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx103</summary><ul>

<li>Passed: 35511/54653 (65.0%)</li>
<li>Failed: 0/54653 (0.0%)</li>
<li>Not Supported: 19142/54653 (35.0%)</li>
</ul></details>


</ul></details>

**Review by s-perron — APPROVED** [review]

**Review by amdrexu — APPROVED** [review]

**Pull request merged**